### PR TITLE
[jenkins][windows] don't remove previous compiled version of ffmpeg & libdvd if nothing changed

### PR DESCRIPTION
**Pull request by Rechi**
## Description
Copy the behaviour for depends on other platform for ffmpeg and libdvd.
They are treated independent so it could happen that non, only one or both are compiled.
If Configuration is set to Release they always get rebuild.

## Motivation and Context
speed up compile time on jenkins for windows

## How Has This Been Tested?
locally tested with different parameters

## Types of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] My code follows the [Code guidelines](https://codedocs.xyz/xbmc/xbmc/code_guidelines.html) of this project 
- [ ] My change requires a change to the documentation, either Doxygen or wiki
- [ ] I have updated the documentation accordingly
- [x] I have read the [CONTRIBUTING](https://github.com/xbmc/xbmc/blob/master/CONTRIBUTING.md) document
- [ ] I have added tests to cover my change
- [ ] All new and existing tests passed

### CODE REVIEW

**Comment by afedchin** [comment]
the ffmpeg depends on gnutls package if it's changed then ffmpeg must be recompiled.
see https://github.com/xbmc/xbmc/blob/master/project/BuildDependencies/DownloadMingwBuildEnv.bat#L32

**Comment by Rechi** [comment]
@afedchin adding `project/BuildDependencies/DownloadMingwBuildEnv.bat` to the checked files/directories should be enough?

**Comment by Rechi** [comment]
added `project/BuildDependencies/DownloadMingwBuildEnv.bat` to checked files/directories

**Comment by Paxxi** [comment]
I can't really judge this, I like the idea but maybe @wsnipex is a better judge if the script is sound?

File: tools/buildsteps/win32/buildffmpeg.sh [hunk 1]
```diff
@@ -12,6 +12,17 @@ FFMPEG_TARGET_OS=mingw32
 do_loaddeps $FFMPEG_VERSION_FILE
 FFMPEGDESTDIR=/xbmc/lib/win32/$LIBNAME
 
+if [ "$(pathChanged $FFMPEGDESTDIR $FFMPEG_VERSION_FILE /xbmc/project/BuildDependencies/DownloadMingwBuildEnv.bat /xbmc/tools/buildsteps/win32)" == "0" ]
```
Comment (wsnipex) [inline tools/buildsteps/win32/buildffmpeg.sh]: pathChanged only tests the first param. So you effectively only check if $FFMPEGDESTDIR changed.
You either need a loop here on in the pathChanged function

**Comment by wsnipex** [comment]
looks ok to me.

**Review by afedchin — APPROVED** [review]

**Comment by Rechi** [comment]
jenkins build and merge

**Pull request merged**